packages/labs/react/src/create-component.ts

```diff
@@ -286,32 +286,29 @@ export function createComponent<
      *
      */
     override render() {
-      // Since refs only get fulfilled once, pass a new one if the user's
-      // ref changed. This allows refs to be fulfilled as expected, going from
+      const {__forwardedRef, ...userProps} = this.props;
```

Comment:
This is pretty clever, could use a comment:
```suggestion
      // Extract and remove __forwardedRef from userProps in a rename-safe way
      const {__forwardedRef, ...userProps} = this.props;
```

Reply:
added the comment